cmd/main.go

```diff
@@ -455,6 +473,104 @@ func run(mainConfig Config) (err error) {
 	return nil
 }
 
+func staticRun(mainConfig Config) (err error) {
+	trustDomain, err := spiffeid.TrustDomainFromString(mainConfig.ctrlConfig.TrustDomain)
+	if err != nil {
+		setupLog.Error(err, "invalid trust domain name")
+		return err
+	}
+
+	ctx := ctrl.SetupSignalHandler()
+
+	setupLog.Info("Dialing SPIRE Server socket")
+	spireClient, err := spireapi.DialSocket(mainConfig.ctrlConfig.SPIREServerSocketPath)
+	if err != nil {
+		setupLog.Error(err, "unable to dial SPIRE Server socket")
+		return err
+	}
+	defer spireClient.Close()
+
+	var mgr ctrl.Manager
+	if mainConfig.ctrlConfig.StaticManifestPath != nil {
+		mgr, err = ctrl.NewManager(&rest.Config{}, mainConfig.options)
+	} else {
+		mgr, err = ctrl.NewManager(ctrl.GetConfigOrDie(), mainConfig.options)
+	}
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		return err
+	}
+	var entryReconciler reconciler.Reconciler
+	if mainConfig.reconcile.ClusterSPIFFEIDs || mainConfig.reconcile.ClusterStaticEntries {
+		entryReconciler = spireentry.Reconciler(spireentry.ReconcilerConfig{
+			TrustDomain:          trustDomain,
+			ClusterName:          mainConfig.ctrlConfig.ClusterName,
+			ClusterDomain:        mainConfig.ctrlConfig.ClusterDomain,
+			K8sClient:            nil,
+			EntryClient:          spireClient,
+			IgnoreNamespaces:     mainConfig.ignoreNamespacesRegex,
+			GCInterval:           mainConfig.ctrlConfig.GCInterval,
+			ClassName:            mainConfig.ctrlConfig.ClassName,
+			WatchClassless:       mainConfig.ctrlConfig.WatchClassless,
+			ParentIDTemplate:     mainConfig.parentIDTemplate,
+			Reconcile:            mainConfig.reconcile,
+			EntryIDPrefix:        mainConfig.ctrlConfig.EntryIDPrefix,
+			EntryIDPrefixCleanup: mainConfig.ctrlConfig.EntryIDPrefixCleanup,
+			StaticManifestPath:   mainConfig.ctrlConfig.StaticManifestPath,
+		})
+	}
+	var federationRelationshipReconciler reconciler.Reconciler
+	if mainConfig.reconcile.ClusterFederatedTrustDomains {
+		federationRelationshipReconciler = spirefederationrelationship.Reconciler(spirefederationrelationship.ReconcilerConfig{
+			K8sClient:          nil,
+			TrustDomainClient:  spireClient,
+			GCInterval:         mainConfig.ctrlConfig.GCInterval,
+			ClassName:          mainConfig.ctrlConfig.ClassName,
+			WatchClassless:     mainConfig.ctrlConfig.WatchClassless,
+			StaticManifestPath: mainConfig.ctrlConfig.StaticManifestPath,
+		})
+		if mainConfig.ctrlConfig.StaticManifestPath == nil {
+			if err = (&controller.ClusterFederatedTrustDomainReconciler{
+				Client:    mgr.GetClient(),
+				Scheme:    mgr.GetScheme(),
+				Triggerer: federationRelationshipReconciler,
+			}).SetupWithManager(mgr); err != nil {
+				setupLog.Error(err, "unable to create controller", "controller", "ClusterFederatedTrustDomain")
+				return err
+			}
+		}
+	}
+	go func() {
+		err = entryReconciler.Run(context.TODO())
```

Comment:
should we run this only when entryReconciler is initialized? maybe using a pointer here?
Same for federation

Reply:
can you use the same context for everything? the one from `ctx := ctrl.SetupSignalHandler()` so if that on is canceled all finish as expected

Reply:
> should we run this only when entryReconciler is initialized? maybe using a pointer here?
> Same for federation

Hmm.... without the interleaved k8s code in the staticRun function, I think we can just move it up into the conditionals above to guard that. Done